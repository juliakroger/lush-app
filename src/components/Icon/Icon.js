import React from 'react'
import * as Icons from '../../svgs'

export const ICONS_TYPES = Object.keys(Icons).map(_icon =>
  _icon.toLocaleLowerCase()
)

const getIcon = type =>
  Object.keys(Icons).find(_icon => _icon.toLocaleLowerCase() === type)

const Icon = ({ type }) => {
  return <span>{React.createElement(Icons[getIcon(type)])}</span>
}

export default Icon
