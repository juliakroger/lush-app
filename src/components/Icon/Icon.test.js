import React from 'react'
import { shallow } from 'enzyme'
import Icon from './Icon'

describe('Component - Icon', () => {
  it('should render Amex svg', () => {
    const component = shallow(<Icon type='amex' />)
    expect(component).toMatchSnapshot()
  })

  it('should render Github svg', () => {
    const component = shallow(<Icon type='github' />)
    expect(component).toMatchSnapshot()
  })

  it('should render Heart svg', () => {
    const component = shallow(<Icon type='heart' />)
    expect(component).toMatchSnapshot()
  })

  it('should render Instagram svg', () => {
    const component = shallow(<Icon type='instagram' />)
    expect(component).toMatchSnapshot()
  })

  it('should render Mastercard svg', () => {
    const component = shallow(<Icon type='mastercard' />)
    expect(component).toMatchSnapshot()
  })

  it('should render Paypal svg', () => {
    const component = shallow(<Icon type='paypal' />)
    expect(component).toMatchSnapshot()
  })

  it('should render Twitter svg', () => {
    const component = shallow(<Icon type='twitter' />)
    expect(component).toMatchSnapshot()
  })

  it('should render Visa svg', () => {
    const component = shallow(<Icon type='visa' />)
    expect(component).toMatchSnapshot()
  })
})
