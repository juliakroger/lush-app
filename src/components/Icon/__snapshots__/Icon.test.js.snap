// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Icon should render Amex svg 1`] = `
<span>
  <Amex />
</span>
`;

exports[`Component - Icon should render Github svg 1`] = `
<span>
  <Github />
</span>
`;

exports[`Component - Icon should render Heart svg 1`] = `
<span>
  <Heart />
</span>
`;

exports[`Component - Icon should render Instagram svg 1`] = `
<span>
  <Instagram />
</span>
`;

exports[`Component - Icon should render Mastercard svg 1`] = `
<span>
  <MasterCard />
</span>
`;

exports[`Component - Icon should render Paypal svg 1`] = `
<span>
  <Paypal />
</span>
`;

exports[`Component - Icon should render Twitter svg 1`] = `
<span>
  <Twitter />
</span>
`;

exports[`Component - Icon should render Visa svg 1`] = `
<span>
  <Visa />
</span>
`;
